### ISSUE TITLE
Use smaller backing tensors when possible

### ISSUE THREAD

**Issue by mortendahl** [mortendahl/tf-encrypted#193]
It might for instance be possible to back `PrimeTensor` by tf.int16 or even tf.int8 in some case, potentially saving on both memory and networking.

**Comment by mortendahl** [comment]
cc @justin1121 something you could be interested in looking at?

**Comment by justin1121** [comment]
Yep. Sounds like all we'll need to do is take an int type as a parameter when creating the factory. Some tf operations might not work with those smaller int types would be something we might want to document in the future.

**Comment by mortendahl** [comment]
maybe it's enough just to look at the prime tensor for now, and eg pick the tf dtype depending on the prime..?

**Comment by mortendahl** [comment]
good point re some operations potentially not working
